plugins/backstage-plugin-coder/src/components/CoderProvider/CoderAuthProvider.tsx

```diff
@@ -99,19 +98,35 @@ export function useCoderAuth(): CoderAuth {
 type CoderAuthProviderProps = Readonly<PropsWithChildren<unknown>>;
 
 export const CoderAuthProvider = ({ children }: CoderAuthProviderProps) => {
-  const identity = useApi(identityApiRef);
-  const { baseUrl } = useBackstageEndpoints();
+  const identityApi = useApi(identityApiRef);
   const [isInsideGracePeriod, setIsInsideGracePeriod] = useState(true);
+  const { api: urlSyncApi } = useUrlSync();
 
   // Need to split hairs, because the query object can be disabled. Only want to
   // expose the initializing state if the app mounts with a token already in
   // localStorage
   const [authToken, setAuthToken] = useState(readAuthToken);
   const [readonlyInitialAuthToken] = useState(authToken);
 
-  const authValidityQuery = useQuery({
-    ...authValidation({ baseUrl, authToken, identity }),
+  const queryIsEnabled = authToken !== '';
+  const authValidityQuery = useQuery<boolean>({
+    queryKey: [...sharedAuthQueryKey, authToken],
+    enabled: queryIsEnabled,
+    keepPreviousData: queryIsEnabled,
     refetchOnWindowFocus: query => query.state.data !== false,
+    queryFn: async () => {
+      // In this case, the request doesn't actually matter. Just need to make any
+      // kind of dummy request to validate the auth
+      const requestInit = await getCoderApiRequestInit(authToken, identityApi);
+      const apiEndpoint = await urlSyncApi.getApiEndpoint();
+      const response = await fetch(`${apiEndpoint}/users/me`, requestInit);
+
+      if (response.status >= 400 && response.status !== 401) {
+        throw new BackstageHttpError('Failed to complete request', response);
+      }
+
+      return response.status !== 401;
```

Comment:
This boolean indicates whether the token is valid, right?

1. Maybe safer to explicitly check for 20x rather than not 401 (I know we return on >=400 above, but not sure if we can get 30x codes, maybe if there is some kind of infinite redirect misconfiguration, and maybe there are very weird cases where a misconfiguration returns a 10x for some reason, and who knows what other codes might be added in the future).
2. That the response includes a user, to make sure we really did hit the deployment and not just some other thing that returns a 20x (for example suppose there is temporarily an HTML page here that returns 200 because the deployment is down for maintenance).

Reply:
Yeah, good call – I think it'll be easier to make this change for the `CoderClient` PR, so I'll make the changes there